ddl/index.go

```diff
@@ -2043,30 +2045,65 @@ func (w *worker) executeDistGlobalTask(reorgInfo *reorgInfo) error {
 
 	taskType := BackfillTaskType
 	taskKey := fmt.Sprintf("ddl/%s/%d", taskType, reorgInfo.Job.ID)
-	elemIDs := make([]int64, 0, len(reorgInfo.elements))
-	for _, elem := range reorgInfo.elements {
-		elemIDs = append(elemIDs, elem.ID)
-	}
+	g, ctx := errgroup.WithContext(context.Background())
+	done := make(chan struct{})
 
-	taskMeta := &BackfillGlobalMeta{
-		Job:             *reorgInfo.Job.Clone(),
-		EleIDs:          elemIDs,
-		EleTypeKey:      reorgInfo.currElement.TypeKey,
-		CloudStorageURI: variable.CloudStorageURI.Load(),
+	// for resuming add index task.
+	taskManager, err := storage.GetTaskManager()
+	if err != nil {
+		return err
 	}
-
-	metaData, err := json.Marshal(taskMeta)
+	task, err := taskManager.GetGlobalTaskByKey(taskKey)
 	if err != nil {
 		return err
 	}
+	if task != nil {
+		g.Go(func() error {
+			defer close(done)
+			backoffer := backoff.NewExponential(dispatcher.RetrySQLInterval, 2, dispatcher.RetrySQLMaxInterval)
+			err := handle.RunWithRetry(ctx, dispatcher.RetrySQLTimes, backoffer, logutil.BgLogger(),
+				func(ctx context.Context) (bool, error) {
+					return true, handle.ResumeTask(taskKey)
+				},
+			)
+			if err != nil {
+				return err
+			}
+			err = handle.WaitGlobalTask(ctx, task)
+			if w.isReorgPaused(reorgInfo.Job.ID) {
```

Comment:
How about the case: we are pausing when the task has `TaskStateSucceed`.

Reply:
> TaskStateSucceed

These transform is forbidden by `func (d *BaseDispatcher) VerifyTaskStateTransform(from, to string) bool`

Reply:
I mean there is a time difference between the determination of `WaitGlobalTask ` and `isReorgPaused` functions
* The global task is finished (real situation)
* We receive a pause DDL operation which means `w.isReorgPaused(reorgInfo.Job.ID) ` is false (This judgment may be delayed)